src/snowflake/snowpark/_internal/analyzer/select_statement.py

```diff
@@ -864,6 +864,16 @@ def sql_query(self) -> str:
     def query_params(self) -> Optional[Sequence[Any]]:
         return self.from_.query_params
 
+    @property
+    def attributes(self) -> Optional[List[Attribute]]:
+        return self._attributes
+
+    @attributes.setter
+    def attributes(self, value: Optional[List[Attribute]]):
+        self._attributes = value
+        if self._session.reduce_describe_query_enabled and value is not None:
```

Comment:
just use this parameter to guard the change?

Reply:
yep